src/snap/notificationObserver.ts

```diff
@@ -0,0 +1,33 @@
+import { DappeteerPage } from "../page";
+import { clickOnElement, openProfileDropdown } from "../helpers";
+import { NotificationList } from "./types";
+
+export const notificationObserver =
```

Comment:
If we have waitForNotifications, why do we need this method? 
Modifying the return argument after it's returned is such a weird concept

Reply:
@mpetrunic sorry I should have opened the draft PR, I wanted to validate the idea with observable. The return argument will change as soon as we have any new state changes inside the method. Imagine it as hooks in React js which is the trendy concept. Indeed we could do the task this way:

```
invokeMethod(notification)
waitForAmountOfNotifications(1)
const notifications = getAllNotifications()
```

The question is only if we would agree to make such an API. As a user might be confused to write the code if some notifications will exist from the previous test cases, then we should wait for a (new notification + previous)

Ideally to have one method notificationObserver without "waitForAmountOfNotifications", but we don't know how long we should await for the Node to be rendered in the DOM. The emitter in this case can serve as a mechanism to await but I'm not sure yet. The fact that the observer is disappearing if we open a new page, we could solve by opening notifications in a new tab

Reply:
> The question is only if we would agree to make such an API. As a user might be confused to write the code if some notifications will exist from the previous test cases, then we should wait for a (new notification + previous)

I agree it is a weird way to do things and method should be called `waitForTotalAmountOfNotifications` and accept timeout as param^^

> The return argument will change as soon as we have any new state changes inside the method. Imagine it as hooks in React js which is the trendy concept. Indeed we could do the task this way:

While that is convenient in react, it's quite unprecedented and uncommon in testing and library development. I'm still leaning more toward a more common event-based system. Maybe you should do a poll in chainsafe engineering slack.

>Ideally to have one method notificationObserver without "waitForAmountOfNotifications", but we don't know how long we should await for the Node to be rendered in the DOM. The emitter in this case can serve as a mechanism to await but I'm not sure yet. The fact that the observer disappears if we open a new page, we could solve this by opening notifications in a new tab. What do you think about the solution?

I mentioned already that new page should be opened for that.